tools/_setup_generation/step_contributor.py

```diff
@@ -0,0 +1,97 @@
+# ################################################################################
+# Taipy Contributor section generation setup step.
+#
+# A contributor section is generated based on internal and external contributors
+# retrieved using GitHub REST APIs.
+# ################################################################################
+import os
+import random
+import requests
+
+from .setup import SetupStep, Setup
+
+
+class ContributorStep(SetupStep):
+
+    def __init__(self):
+        self.BASE_URL = "https://api.github.com"
+        self.ORGANIZATION_URL = f"{self.BASE_URL}/orgs/avaiga"
+        self.MEMBERS_URL = f"{self.ORGANIZATION_URL}/members"
+        self.REPOS = f"{self.ORGANIZATION_URL}/repos"
+        self.REPO_URLS = []
+        self.MEMBERS = {}
+        self.CONTRIBUTORS = {}
+        self.ANONYMOUS = ["dependabot[bot]"]
+        self.PATH = ""
+        self.TEMPLATE_SUFFIX = "_template"
+
+
+    def enter(self, setup: Setup):
+        self.PATH = os.path.join(setup.docs_dir, "credits", "contributors.md")
+
+    def get_id(self) -> str:
+        return "contributors"
+
+    def get_description(self) -> str:
+        return "Generating the contributors"
+
+    def setup(self, setup: Setup) -> None:
+        self.get_repo_urls()
+        self.get_avaiga_members()
+        self.get_contributors()
+        self.build_content((self.MEMBERS, "[AVAIGA_TEAM_MEMBERS]"), (self.CONTRIBUTORS, "[TAIPY_CONTRIBUTORS]"))
+
+    def get_repo_urls(self):
+        repos = requests.get(self.REPOS).json()
```

Comment:
Are you certain that this works for anonymous users?
I mean no access key, no nothing... and still works?

Reply:
I think so. I only get public information that you can get on GitHub directly.